File: app/src/main/java/com/imangazalievm/bubbble/presentation/userprofile/shots/UserShotsView.java
package com.imangazalievm.bubbble.presentation.userprofile.shots;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.imangazalievm.bubbble.domain.global.models.Shot;

import java.util.List;


@StateStrategyType(AddToEndSingleStrategy.class)
public interface UserShotsView extends MvpView {

    void showNewShots(List<Shot> newShots);
    @StateStrategyType(OneExecutionStateStrategy.class)
    void showShotsLoadingProgress();
    void hideShotsLoadingProgress();
    void showShotsLoadingMoreProgress();
    void hideShotsLoadingMoreProgress();
    void openShotDetailsScreen(long shotId);
    void showNoNetworkLayout();
    void hideNoNetworkLayout();
    void showLoadMoreError();

}

File: app/src/main/java/com/bubbble/presentation/userprofile/shots/UserShotsView.kt
package com.bubbble.presentation.userprofile.shots

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import com.bubbble.core.models.Shot
import com.bubbble.presentation.global.mvp.BaseMvpView

@StateStrategyType(AddToEndSingleStrategy::class)
interface UserShotsView : BaseMvpView {

    fun showNewShots(newShots: List<com.bubbble.core.models.Shot>)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun showShotsLoadingProgress()

    fun hideShotsLoadingProgress()

    fun showShotsLoadingMoreProgress()

    fun hideShotsLoadingMoreProgress()

    fun openShotDetailsScreen(shotId: Long)

    fun showNoNetworkLayout()

    fun hideNoNetworkLayout()

    fun showLoadMoreError()

}